slatedb/src/cached_object_store/storage_fs.rs

```diff
@@ -436,28 +439,30 @@ impl FsCacheEvictor {
     // when the evictor is dropped.
     #[allow(clippy::disallowed_methods)]
     pub async fn track_entry_accessed(&self, path: std::path::PathBuf, bytes: usize, evict: bool) {
-        if !self.started().await {
+        if !self.started() {
             return;
         }
 
         self.tx.send((path, bytes, evict)).await.ok();
     }
 }
 
-/// FsCacheEvictorInner manages the cache entries in an in-memory trie, and evict the cache entries
-/// when the cache size exceeds the limit. it uses a pick-of-2 strategy to approximate LRU, and evict
-/// the older file when the cache size exceeds the limit.
-///
-/// On start up, FsCacheEvictorInner will scan the cache folder to load the cache files into the in-memory
-/// trie cache_entries. This loading process is interleaved with the maybe_evict is being called, so the
-/// cache entries should be wrapped with Mutex<_>.
+#[derive(Debug, Clone)]
+struct CacheEntry {
+    access_time: DateTime<Utc>,
+    size_bytes: usize,
+    key_index: usize,
+}
+
+/// Manages cache entries and evicts them when size exceeds the limit using pick-of-2 LRU approximation.
 #[derive(Debug)]
 struct FsCacheEvictorInner {
     root_folder: std::path::PathBuf,
     batch_factor: usize,
     max_cache_size_bytes: usize,
     track_lock: Mutex<()>,
-    cache_entries: Mutex<Trie<std::path::PathBuf, (DateTime<Utc>, usize)>>,
+    cache_entries: Mutex<HashMap<std::path::PathBuf, CacheEntry>>,
+    cache_keys: Mutex<Vec<std::path::PathBuf>>,
```

Comment:
does `cache_keys` and `cache_entries` expected to be protected inside a single Mutex?